### PR TITLE
MM-29571: Implement basic support

### PR DESCRIPTION
**Pull request by agnivade**

### CODE REVIEW

**Comment by agnivade** [comment]
This is dependent on https://mattermost.atlassian.net/browse/MM-29569. So I'll need to make the changes once that goes in. But the basic functionality can be reviewed.

**Comment by agnivade** [comment]
Still needs the mlog changes though.

File: internal/server/handler_test.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	}
+
+	s := &Server{
+		logger: log.New(os.Stderr, "", log.Lshortfile|log.LstdFlags),
+		cfg:    cfg,
+	}
+
+	w := httptest.NewRecorder()
+
+	s.writeError(w, errors.New("error from valhalla"))
```
Comment (isacikgoz) [inline internal/server/handler_test.go:128]: 😂

File: internal/server/server.go [hunk 2]
```diff
@@ -20,14 +24,38 @@ var (
...
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+				DualStack: true,
+			}).DialContext,
+			ForceAttemptHTTP2:     true,
+			MaxConnsPerHost:       cfg.ServiceSettings.MaxConnsPerHost,
+			MaxIdleConns:          100,
+			IdleConnTimeout:       90 * time.Second,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ResponseHeaderTimeout: 30 * time.Second,
```
Comment (isacikgoz) [inline internal/server/server.go:52]:
```suggestion
			ResponseHeaderTimeout: time.Duration(cfg.ServiceSettings.ResponseHeaderTimeout) * time.Second,
```

**Review by isacikgoz — APPROVED** [review]
LGTM with a minor suggestion.

File: internal/server/server.go (hunk 2 above)
Comment (agnivade) [reply internal/server/server.go:52]: Yeah, I didn't really see a need for response header timeouts to be configurable. Hence didn't keep it. But we can include it if we need it later.

**Pull request merged**